### PR TITLE
Added running WASM within the zkvm example

### PR DESCRIPTION
**Pull request by mothran**
Added a example the runs the supplied WASM bytecode within the zkvm and commits the results. Using a simple `fib()` WAT module as an example function.

### CODE REVIEW

File: examples/wasm/methods/guest/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    let wasm: Vec<u8> = env::read();
+    let iters: i32 = env::read();
+
+    // Derived from the wasmi example: https://docs.rs/wasmi/0.29.0/wasmi/#example
+    let module = Module::new(&engine, &mut &wasm[..]).expect("Failed to create module");
+    type HostState = u32;
+
+    let linker = <Linker<HostState>>::new(&engine);
+    let mut store = Store::new(&engine, 42);
```
Comment (tzerrell) [inline examples/wasm/methods/guest/src/main.rs:35]: Why does this store `42`? This feels like it may be an accidental relic of the source example...

**Review by tzerrell — APPROVED** [review]
Awesome!

**Review by flaub — APPROVED** [review]

File: examples/wasm/methods/guest/src/main.rs (hunk 1 above)
Comment (mothran) [reply examples/wasm/methods/guest/src/main.rs:35]: Yeah I think its an artifact of https://docs.rs/wasmi/latest/wasmi/#example and it requires some data to be passed with the correct type: https://docs.rs/wasmi/latest/wasmi/struct.Store.html#method.new

Comment (tzerrell) [reply examples/wasm/methods/guest/src/main.rs:35]: :+1: gotcha, so the value is arbitrary and unused, but having _something_ there is required.

**Pull request merged**